bin/publish.ml

```diff
@@ -120,13 +118,15 @@ let artefacts =
     | s when String.is_prefix ~affix:alt_prefix s -> (
         match String.(with_range ~first:(length alt_prefix) s) with
         | "" -> `Error "`alt-' alternative artefact kind is missing"
-        | kind -> `Ok (`Alt kind) )
+        | kind ->
+            App_log.unhappy (fun l -> l Deprecate_delegates.warning_usage);
+            `Ok (`Alt kind) )
     | s -> `Error (strf "`%s' unknown publication artefact" s)
   in
   let printer ppf = function
     | `Doc -> Fmt.string ppf "doc"
     | `Distrib -> Fmt.string ppf "distrib"
-    | `Alt a -> Fmt.pf ppf "alt-%s" a
+    | `Alt a -> Fmt.pf ppf Deprecate_delegates.alt_artefacts_pp a
```

Comment:
This is just used by Cmdliner to report errors I think, not sure it's worth redifining the printer.

Reply:
The idea was to make removing delegates easier: when removing delegates we'd get a compilation error here when removing the `Deprecate_delegates` warning and wouldn't forget about this match arm. But I've reverted this.